### PR TITLE
Show Annotation Segment Counts in Dashboard

### LINKED ISSUE #7431: Add segment counts to dashboard
## Detailed Description
Similar to the skeleton statistics (node count, tree count etc) we should show the segment count of an annotation/task in the various lists and views:

- Dashboard > Annotations
- Admin > Tasks > Sub-View for the annotations per Task 
- (Maybe in the right-hand side info panel of an annotation)

This requires a little bit of frontend work but mostly backend. The segment counts could be cached using a similar structure as for the skeleton nodes.

## Context

* [ ] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
* [ ] Specific to webknossos.org (set `isDemoInstance=true` in `application.conf`)

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://savesegmentstats.webknossos.xyz

### Steps to test:
- Create annotation, annotate some, dashboard should show tree + segment count
 
### TODOs:
- [x] Frontend
  - [x] When saving volume layer updates, send stats object with current segmentCount
  - [X] Adapt dashboard to aggregate stats for annotation from annotation layers
    - [x] How to display segment count? new icon? do we need more space?
    - ~[ ] Show infos for all layers in tooltip?~ Philipp: Not important for now
- [x] Backend
  - [x] Store stats per annotation layer
  - [x] SQL migration
  - [x] Include stats in annotation layer json
  - [x] How to encode unknown status? empty object?
  - [x] Initialize stats
    - [x] as unknown (`{}` after merge, create from task, upload)
    - [x] as zeroed when creating empty annotation layer

### Issues:
- fixes #7431

------
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)

### CODE REVIEW

**Comment by philippotto** [comment]
@fm3 The frontend should send the segmentCount now. The dashboard receives `stats: {}` for the annotations now, but I guess the backend needs to be adapted.

**Comment by fm3** [comment]
Great, thanks! Yes, I moved the stats to the annotation layers inside of the annotation json. They should :tm: already be filled, but I did not test this end-to-end yet.

**Comment by fm3** [comment]
Yeah, seems to work :)

image

Note that the stats in the layers may also still be `{}` for “unknown” (I plan to initialize it with zeroes but I think it’s not easy to get the real values after upload or merge, so I probably won’t do those in this PR).

Do you think the dashboard frontend code can work with this new format?

**Comment by philippotto** [comment]
Cool, I adapted the dashboard now. What's up with the old `stats` property in the actual annotation? Is that always empty now? Should we remove it?

**Comment by fm3** [comment]
Yes, it’s always empty now. Removing it completely may break old libs clients. Feel free to remove it from the frontend types. I plan to remove it from the json in a later version.

**Comment by fm3** [comment]
@philippotto thanks for taking care of the frontend!

Do you think it would be possible to hide the segments if the annotation has no volume layer, and hide the tree count if the annotation has no skeleton layer? This way the user could notice at a glance what kind of annotation they have? (I’m also thinking that we might want to remove the skeleton/volume/hybrid tag from the dashboard in the future, because especially “hybrid” is explained nowhere)

I think the backend is ready for review, feel free to request reviews when you think this is done :)

**Comment by philippotto** [comment]
> Do you think it would be possible to hide the segments if the annotation has no volume layer, and hide the tree count if the annotation has no skeleton layer? This way the user could notice at a glance what kind of annotation they have? (I’m also thinking that we might want to remove the skeleton/volume/hybrid tag from the dashboard in the future, because especially “hybrid” is explained nowhere)

Yes, good points :+1: It's done now.

File: frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts
Comment (MichaelBuessemeyer) [inline frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts:69]: Maybe make this an own (local) type, to avoid repetition in lines 98-104?

**Review by MichaelBuessemeyer — COMMENTED** [review]
The frontend code looks very good :+1:. I came up with one suggestion to make the code DRYer.
My testing also showed that the feature seems to work seamlessly.
I'd say, the frontend is merge ready :ship: 

@frcroth Could you please approve this PR once everything looks good in the backend part?

**Review by frcroth — APPROVED** [review]
LGTM

**Pull request merged**
